llvm/lib/Target/AIE/AIEHazardRecognizer.h

```diff
@@ -178,12 +178,8 @@ class AIEHazardRecognizer : public ScheduleHazardRecognizer {
   unsigned computeScoreboardDepth() const;
 
 protected:
-  ScheduleHazardRecognizer::HazardType getHazardType(const MCInstrDesc &Desc,
-                                                     MemoryBankBits MemoryBanks,
-                                                     int DeltaCycles);
-  ScheduleHazardRecognizer::HazardType getHazardType(unsigned SchedClass,
-                                                     SlotBits SlotSet,
-                                                     MemoryBankBits MemoryBanks,
+  ScheduleHazardRecognizer::HazardType getHazardType(const MachineInstr *MI,
+                                                     const MCInstrDesc &Desc,
```

Comment:
Please only have `const MachineInstr &MI` and `int DeltaCycles` as parameters. We can easily retrieve `Desc` from `MI`.

Also note I changed the pointer to a reference.